joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/ast/AstForExpressionsCreator.scala

```diff
@@ -456,21 +458,52 @@ trait AstForExpressionsCreator(implicit withSchemaValidation: ValidationMode) {
       else s"$methodFqName:$explicitSignature"
     val (fullName, signature) = typeInfoProvider.fullNameWithSignature(expr, (explicitFullName, explicitSignature))
 
+    val bindingContext = typeInfoProvider.bindingContext
+    val call           = bindingContext.get(BindingContext.CALL, expr.getCalleeExpression)
+    val resolvedCall   = bindingContext.get(BindingContext.RESOLVED_CALL, call)
+
+    val dispatchType =
+      if (resolvedCall == null) {
+        DispatchTypes.STATIC_DISPATCH
+      } else {
+        if (resolvedCall.getDispatchReceiver == null) {
+          DispatchTypes.STATIC_DISPATCH
+        } else {
+          resolvedCall.getResultingDescriptor match {
+            case functionDescriptor: FunctionDescriptor
+                if functionDescriptor.getVisibility == DescriptorVisibilities.PRIVATE =>
+              DispatchTypes.STATIC_DISPATCH
```

Comment:
From my reading of the logic below, the `this` identifier at argument 0 will be missing for calls of `private` instance methods. Or am I missing something there?

Reply:
Good catch. Updated the PR. Please have another look.